diesel/src/connection/transaction_manager.rs

```diff
@@ -114,43 +117,57 @@ where
         let transaction_depth = conn.transaction_state().transaction_depth;
         if transaction_depth <= 1 {
             match conn.batch_execute("COMMIT") {
-                // When any of these kinds of error happen on `COMMIT`, it is expected
-                // that a `ROLLBACK` would succeed, leaving the transaction in a non-broken state.
-                // If there are other such errors, it is fine to add them here.
```

Comment:
We should add this comment to the new `process_commit_error` function, because it makes sense there.